docs/analytic_beam_tutorial.rst

```diff
@@ -0,0 +1,860 @@
+The new beam needs to inherit from either the :class:`pyuvdata.analytic_beam.AnalyticBeam`,
+or the :class:`pyuvdata.analytic_beam.UnpolarizedAnalyticBeam`, which are base
+classes that specify what needs to be defined on the new class. Unpolarized
+beams (based on the ``UnpolarizedAnalyticBeam`` class) have fewer things that
+need to be specified.
+
+Note that while unpolarized beams are simpler to define and think about, they
+are quite unphysical and can have results that may be surprising to radio
+astronomers. Since unpolarized feeds respond equally to all orientations of the
+E-field, if two feeds are specified they will have cross-feed power responses that
+are more similar to typical auto-feed power responses (and they will be identical
+to auto-feed power responses if the two feeds have the same beam shapes).
+
+Setting parameters on the beam
+******************************
+
+If the new beam has any parameters that control the beam response (e.g. diameter),
+The class must have an ``@dataclass`` decorator and the parameters must be listed
+in the class definitions with type annotations and optionally defaults (these
+are called ``fields`` in the dataclass, see the examples below and
+`dataclass <https://docs.python.org/3/library/dataclasses.html>`_ for more details).
+
+If you need to do some manipulation or validation of the parameters after they
+are specified by the user, you can use the dataclass's ``__post_init__`` method
+to do that, being sure to call the super class ``__post_init__`` as well.
+The gaussian beam example below shows how this can be done.
+
+Polarized beams
+***************
+
+For polarized beams (based on the ``AnalyticBeam`` class), the following items
+may be specified, the defaults on the ``AnalyticBeam`` class are noted:
+
+  - ``feed_array``: This an array of feed strings (a list can also be passed,
+    it will be converted to an array). The default is ``["x", "y"]``.
+    This is a a dataclass field, so if it is specified, the class must have
+    ``@dataclass`` decorator and it should be specified with type annotations
+    and optionally a default (see examples below).
+
+  - ``x_orientation``: For linear polarization feeds, this specifies what the
+    ``x`` feed polarization correspond to, allowed values are ``"east"`` or
+    ``"north"``, the default is ``"east"``. Should be set to ``None`` for
+    circularly polarized feeds.
+    This is a a dataclass field, so if it is specified, the class must have
+    ``@dataclass`` decorator and it should be specified with type annotations
+    and optionally a default (see examples below).
+
+  - ``basis_vector_type``: This defines the coordinate system for the
+    polarization basis vectors, the default is ``"az_za"``. Currently only
+    ``"az_za"`` is supported, which specifies that there are 2 vector directions
+    (i.e. ``Naxes_vec`` is 2).
+    This should be defined as a class variable (see examples below).
```

Comment:
I don't think we need to mention these -- these are supposed to be the same for all subclasses, so shouldn't be over-written by someone writing a new beam subclass.

Reply:
No, there's a warning if it's not defined on a beam that's based on the `AnalyticBeam` class. We have it there so we can support other basis vectors in the future as needed.

Reply:
Ah yes, that's correct.